scripts/gha/lint_commenter.py

```diff
@@ -0,0 +1,280 @@
+      [args.curl,
+       '-s', '-X', 'GET',
+       '-H', 'Accept: application/vnd.github.v3+json',
+       '-H', 'Authorization: token %s' % args.token,
+       request_url
+      ] + ([] if not args.verbose else ['-v'])).decode('utf-8').rstrip('\n'))
+  commit_sha = pr_data['head']['sha']
+  skip_lint = False
+  if 'labels' in pr_data:
+    for label in pr_data['labels']:
+      if label['name'] == LABEL_TO_SKIP_LINT:
+        skip_lint = True
+        break
+
+  if skip_lint:
+    print('PR #%s has "%s" label, skipping lint checks' % (args.pr_number, LABEL_TO_SKIP_LINT))
+    exit(0)
+
+  if args.verbose:
+    print('Commit sha:', commit_sha)
+
+  # Get the diff for the pull request.
+  # GET /repos/{owner}/{repo}/pulls/{pull_number}
+  request_url = 'https://api.github.com/repos/%s/%s/pulls/%s' % (repo_owner, repo_name, args.pr_number)
+  header = 'Accept: application/vnd.github.VERSION.diff'
+  if args.verbose:
+    print('request_url: %s' % request_url)
+  pr_diff = subprocess.check_output(
+      [args.curl,
+       '-s', '-o', '-', '-w', '\nHTTP status %{http_code}\n',
+       '-X', 'GET',
+       '-H', header,
+       '-H', 'Authorization: token %s' % args.token,
+       request_url
+      ] + ([] if not args.verbose else ['-v'])).decode('utf-8')
+  # Parse the diff to determine the position (within the PR) of each source line.
+  # Only lint lines that refer to valid positions in the PR will be displayed.
+  # Information on what this means here:
+  # https://docs.github.com/en/rest/reference/pulls#create-a-review-comment-for-a-pull-request
+  line_reference = {}
+  file_list = []
+  pr_patch = PatchSet(pr_diff)
+  for pr_patch_file in pr_patch:
+    # Skip removed files.
+    if pr_patch_file.removed: continue
+    # Skip files that match an EXCLUDE_PATH_REGEX
+    excluded = False
+    for exclude_regex in EXCLUDE_PATH_REGEX:
+      if re.search(exclude_regex, pr_patch_file.path):
+        excluded = True
+        break
+    if excluded: continue
+    file_list.append(pr_patch_file.path)
+    pr_file_position = 1
+    line_reference[pr_patch_file.path] = {}
+    for hunk in pr_patch_file:
+      pr_file_position = pr_file_position + 1
+      for line in range(
+          hunk.target_start,
+          hunk.target_start + hunk.target_length):
+        line_reference[pr_patch_file.path][line] = pr_file_position
+        pr_file_position = pr_file_position + 1
+
+  # Now we have a list of files in repo
+  try:
+    lint_results=subprocess.check_output([
+        args.lint_command,
+        '--output=emacs',
+        ('--filter=%s' % CPPLINT_FILTER),
+        ('--repository=..')
+    ] + file_list, stderr=subprocess.STDOUT).decode('utf-8').split('\n')
+  except subprocess.CalledProcessError as e:
+    # Nothing to do if there is an exception.
+    lint_results=e.output.decode('utf-8').split('\n')
+
+  comments = []
+  for line in lint_results:
+    m = re.match(r'([^:]+):([0-9]+): *(.*[^ ]) +\[[^]]+\] \[\d+\]$', line)
```

Comment:
probably worth a comment as to what this is matching.

Reply:
Done.